packages/plugin-core/src/dendronExtensionInterface.ts

```diff
@@ -0,0 +1,70 @@
+import vscode from "vscode";
+import {
+  DWorkspaceV2,
+  WorkspaceSettings,
+  WorkspaceType,
+} from "@dendronhq/common-all";
+import { IWSUtilsV2 } from "./WSUtilsV2Interface";
+import { IWorkspaceService } from "@dendronhq/engine-server";
+import { IEngineAPIService } from "./services/EngineAPIServiceInterface";
+import { ICommandFactory } from "./commandFactoryInterface";
+
+export type DendronWorkspaceSettings = Partial<{
+  "dendron.dailyJournalDomain": string;
+  "dendron.defaultJournalName": string;
+  "dendron.defaultJournalDateFormat": string;
+  "dendron.defaultJournalAddBehavior": string;
+  "dendron.defaultScratchName": string;
+  "dendron.defaultScratchDateFormat": string;
+  "dendron.defaultScratchAddBehavior": string;
+  "dendron.copyNoteUrlRoot": string;
+  "dendron.linkSelectAutoTitleBehavior": string;
+  "dendron.defaultLookupCreateBehavior": string;
+  "dendron.defaultTimestampDecorationFormat": string;
+  "dendron.rootDir": string;
+  "dendron.dendronDir": string;
+  "dendron.logLevel": string;
+  "dendron.trace.server": string;
+  "dendron.serverPort": string;
+}>;
+
+export interface IDendronExtension {
```

Comment:
Going forward we should use IDendronExtension in places of concrete DendronExtension

Reply:
Can you go ahead and add some comments info for `IDendronExtension`? Doesn't need to be complete, we can build upon the docstrings over time.

Reply:
Good point, added.